src/components/Books.js

```diff
@@ -1,21 +1,31 @@
-import React from 'react';
-import { useSelector } from 'react-redux';
+import React, { useEffect } from 'react';
+import { useSelector, useDispatch } from 'react-redux';
 import Form from './Form';
 import Book from './Book';
 import NavBar from './NavBar';
+import { getBooksAction } from '../redux/books/books';
 
 const Books = () => {
-  const books = useSelector((state) => state.book);
+  const { books, status } = useSelector((state) => state.book);
+  const dispatch = useDispatch();
+
+  useEffect(() => {
+    dispatch(getBooksAction());
+  }, [dispatch]);
+
+  if (status === 'loading') {
+    return <h1>Loading...</h1>;
+  }
+
   return (
     <>
       <NavBar />
       <div className="books-con">
-        {books.map((book) => (
+        {(books).map((book) => (
           <Book
             key={book.id}
             title={book.title}
             author={book.author}
-            category={book.category}
```

Comment:
- [ ] When I reload the page, I get an error message which says id is required but it is undefined. The way you are accessing it here is wrong. It should be `book.item_id` as you defined in your reducers. Kindly implement this change. 👍